draft-tulshibagwale-oauth-transaction-tokens-03.md

```diff
@@ -308,28 +308,50 @@ The JWT body MUST have the following claims:
 
 * An `iss` claim, whose value is a URN {{RFC8141}} that uniquely identifies the workload or the Txn-Token Service that created the Txn-Token.
 * An `iat` claim, whose value is the time at which the Txn-Token was created.
+* An `aud` claim, whose value is a URN {{RFC8141}} that uniquely identifies the audience of the Txn-Token. This MUST identify the trust domain in which the Txn-Token is used.
 * An `exp` claim, whose value is the time at which the Txn-Token expires.
 * A `txn` claim, whose value is the unique transaction identifier as defined in Section 2.2 of {{RFC8417}}. When used in the transaction token, it identifies the entire call chain.
 * A `sub_id` claim, whose value is the unique identifier of the user or workload on whose behalf the call chain is being executed. The format of this claim MAY be a Subject Identifier as specified in {{SubjectIdentifiers}}.
 * An `azc` claim, whose value is a JSON object that contains values that remain constant in the call chain.
 
-{{figleaftxtokenbody}} shows a non-normative example of the JWT body of a Txn-Token:
+### Optional Claims
+The JWT body MAY have the following claims:
+
+#### Originator
+The Txn-Token MAY contain an `orig` claim, whose value is a JSON object the describes the origin of the transaction. This MAY include the IP address information of the originating user, as well as information about the computational entity that requested the Txn-Token.
+
+The JSON value of the `orig` claim MAY include any values the Txn-Token Service determines are interesting to downstream services that rely on the Txn-Token. The following claims are defined so that if they are included, they have the following meaning:
+* `req_ip` The IP address of the requester. This MAY be the end-user or a robotic process that requested the Transaction
+* `authn` The authentication method used to idenitfy the requester. Its value is a URN that uniquely identifies the method used.
+* `req_gw` A URN that uniquely identifies the computational entity that requested the Txn-Token. This entity MUST be within the Trust Domain of the Txn-Token.
```

Comment:
What about `req_wl` for requesting workload or `req_app` for requesting app. My thoughts here are that the requesting entity may not always be a gateway. In my original proposal I called this the `act` claim representing "actor" per the token exchange spec.